go/pkg/cas/upload.go

```diff
@@ -57,6 +69,48 @@ type PathSpec struct {
 	Exclude *regexp.Regexp
 }
 
+func (ps *PathSpec) deepCopy() *PathSpec {
+	ret := *ps
+	if len(ps.Allowlist) > 0 {
+		ret.Allowlist = make([]string, len(ps.Allowlist))
+		copy(ret.Allowlist, ps.Allowlist)
+	}
+	return &ret
+}
+
+// validateAndNormalize validates and normalizes ps.
```

Comment:
Can you add more info here that cannot be inferred from the function name?

Reply:
I wasn't sure about this one. My true intention was to just document that mode must match ps.Path, but Go requires the comment to start with the symbol name. At the same time, I'd like to avoid documenting the implementation details of this func. I.e. I don't we need to document that it returns an error if ps.Path is relative

Ideas?

Reply:
no longer applicable